src/main/java/games/strategy/engine/framework/lookandfeel/LookAndFeel.java

```diff
@@ -87,4 +88,40 @@ public static void setupLookAndFeel() {
     });
   }
 
+  private static String getDefaultLookAndFeel() {
+    String defaultLookAndFeel = SubstanceGraphiteLookAndFeel.class.getName();
+
+    if (SystemProperties.isMac()) {
+      // stay consistent with mac look and feel if we are on a mac
+      defaultLookAndFeel = UIManager.getSystemLookAndFeelClassName();
+    }
+
+    final String userDefault = ClientSetting.LOOK_AND_FEEL_PREF.value();
+    final List<String> availableSkins = getLookAndFeelAvailableList();
+
+    if (availableSkins.contains(userDefault)) {
+      return userDefault;
+    }
+    if (availableSkins.contains(defaultLookAndFeel)) {
+      setDefaultLookAndFeel(defaultLookAndFeel);
+      return defaultLookAndFeel;
+    }
+    return UIManager.getSystemLookAndFeelClassName();
+  }
+
+  private static void setDefaultLookAndFeel(final String lookAndFeelClassName) {
+    try {
+      UIManager.setLookAndFeel(lookAndFeelClassName);
+    } catch (ClassNotFoundException | InstantiationException | IllegalAccessException
+        | UnsupportedLookAndFeelException e) {
+      ClientLogger.logError("Unable to load look and feel: " + lookAndFeelClassName
+          + ", retaining the old look and feel. Please do not select this look and feel, it does not work."
+          + " Please do report this to the developers so the look and feel can be addressed. When doing so, please"
+          + " include this list of installed look and feel debug data: "
+          + Arrays.asList(UIManager.getInstalledLookAndFeels()), e);
```

Comment:
You should use Arrays.toString instead of Arrays.asList 
This will just results in java.util.ArrayList@<hashcode>

Reply:
@RoiEXLab, can you elaborate on what your concern is here?  When I run the following code:

```java
System.out.println(Arrays.asList(UIManager.getInstalledLookAndFeels()));
System.out.println(Arrays.toString(UIManager.getInstalledLookAndFeels()));
```

I see identical output.

It looks like `AbstractCollection` (which is an ancestor of `ArrayList`) overrides `toString()` with an equivalent algorithm to `Arrays#toString()`.

If your concern is that you can avoid the creation of a new `ArrayList` by using `Arrays#toString()`, I'll buy that. :smile:

Reply:
I could have sworn that ArrayList#toString has the default toString implementation of Object. Good to know! 
But still, we don't need to create an extra object if we just want to print one here.